src/screens/SearchScreen.tsx

```diff
@@ -24,19 +19,20 @@ const SearchScreen = () => {
       {errorMessage ? <Text>{errorMessage}</Text> : null}
       <ScrollView>
         <ResultsList
-          results={filterResultsByPrice("€")}
+          results={filterResultsByPrice(results, "€")}
```

Comment:
The filtering should be done in `useResults`. We want view components to be as dumb as possible, and leave all calculations outside.